showkase-component-impressions/src/main/java/com/vinaygaba/showkase_component_impressions/ComponentImpressions.kt

```diff
@@ -0,0 +1,138 @@
+package com.vinaygaba.showkase_component_impressions
+
+import android.annotation.SuppressLint
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.composed
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.platform.LocalView
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.collect
+
+/**
+ * Passed in the callback of the [visibilityEvents] Modifier. Contains information about the
+ * visibility of a given composable function that uses the visibilityEvents Modifier.
+ *
+ * @param key The unique identifier that you passed with the [visibilityEvents] modifier.
+ * @param visibilityPercentage A value between 0.0 and 1.0 that represents how much a given composable
+ * was visible on the screen. 0.0 represents that it wasn't visible on the screen and 1.0 represents
+ * that it was completely visible on the screen.
+ * @param boundsInWindow A [Rect] object that contains information about the bounds of a given
+ * composable. When the bounds are (0, 0, 0, 0), it means that the composable was not visible on
```

Comment:
> When the bounds are (0, 0, 0, 0), it means that the composable was not visible on screen.

could use a sealed class instead to represent this in the type system more clearly